### PR TITLE
autoAddBoardId should always be defined as "none" or board_id

### PR DESCRIPTION
**Pull request by maryhipp**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Have you discussed this change with the InvokeAI team?
- [ ] Yes
- [ ] No, because:

      
## Have you updated all relevant documentation?
- [ ] Yes
- [ ] No

## Description
I'm not positive that I did this right - but I was running into issues locally where `autoAddBoardId` was `null`, which it seems is an unexpected state. It was getting sent as a `board_id` param when creating an upload in the "uncategorized" view. It seems that `autoAddBoardId` should always be defined, and that if a board is not selected, it should always be "none" - never null or undefined.

## Related Tickets & Documents

- Related Issue #
- Closes #

## Added/updated tests?

- [ ] Yes
- [ ] No : _please replace this line with details on why tests
      have not been included_

### CODE REVIEW

**Comment by maryhipp** [comment]
@KevinBrack - in case you want to take a look, may overlap with your recent work on the checkbox setting

File: invokeai/frontend/web/src/features/gallery/store/gallerySlice.ts [hunk 1]
```diff
@@ -41,6 +41,10 @@ export const gallerySlice = createSlice({
       state.galleryView = 'images';
     },
     autoAddBoardIdChanged: (state, action: PayloadAction<BoardId>) => {
+      if (!action.payload) {
```
Comment (maryhipp) [inline invokeai/frontend/web/src/features/gallery/store/gallerySlice.ts:44]: I couldn't actually even find where this being set to `null` - the initial state has it defined as "none"

**Comment by maryhipp** [comment]
This begs the bigger question - should `autoAddToBoardId` always be assigned? If you have this "off", isn't that the same as having it on and set to "none"?

**Comment by psychedelicious** [comment]
You're right, it should never be null or undefined. Maybe you had leftover state from a prev version? I've also noticed some weirdness with hot module reloading and the RTK cache. 

Originally I used `undefined` to mean "put the image in Uncategorized", but was getting mixed up in the board logic. Making it either "none" or the actual board ID made more sense to me, but if this is confusing we can change it back. Just need to be careful in the query definitions.

**Comment by psychedelicious** [comment]
The remaining places where we are checking if the board id is falsy are from me missing those spots when rebasing the multi select pr. I'll do another pass and update this pr.

File: invokeai/frontend/web/src/features/gallery/store/gallerySlice.ts (hunk 1 above)
Comment (psychedelicious) [reply invokeai/frontend/web/src/features/gallery/store/gallerySlice.ts:44]: I just reviewed every instance of `autoAddBoardId`, the action that sets it, and any times it is set directly in the `gallerySlice` reducers. Cannot find anywhere that would result in it being set with `null` or `undefined`. Weird...

**Review by psychedelicious — APPROVED** [review]
Thanks for catching this andd hopefully `autoAddBoardId` is never not a string again

**Comment by maryhipp** [comment]
Definitely ~possible~ probable that my local storage could have something to do with it

**Comment by KevinBrack** [comment]
NoBoardBoard.tsx was using board undefined here. When I set up auto add board on click I used the same syntax that was there. Is this supposed to be 'none'?

```
const NoBoardBoard = memo(({ isSelected }: Props) => {
  const dispatch = useAppDispatch();
  const { autoAddBoardId } = useAppSelector(selector);
  const { autoAddBoardId, autoAssignBoardOnClick, isProcessing } =
    useAppSelector(selector);
  const boardName = useBoardName(undefined);
  const handleSelectBoard = useCallback(() => {
    dispatch(boardIdSelected(undefined));
  }, [dispatch]);
    if (autoAssignBoardOnClick && !isProcessing) {
      dispatch(autoAddBoardIdChanged(undefined));
    }
  }, [dispatch, autoAssignBoardOnClick, isProcessing]);
  ```

**Comment by psychedelicious** [comment]
@KevinBrack No, you're good - it was supposed to be `undefined` in your PR. I updated your code in `NoBoardBoard`'s to use 'none' in the multi-select PR, which changed it to 'none', so that wasn't an issue

**Pull request merged**